tests/system/common.py

```diff
@@ -800,7 +801,49 @@ def task_by_name(tasks, name):
         if task['name'] == name:
             return task
 
-    assert False, "Did not find task with name %s in this list of tasks: %s" % (name, tasks,)
+    assert False, "Did not find task with name {} in this list of tasks: {}".format(name, tasks)
+
+
+def assert_app_in_all_domains(app, regions=None, zones=None):
+    """Asserts that all tasks of the given app are distributed in all of the given region and/or zone
```

Comment:
Hm, does this mean we assert that all zones and regions are unique of an app deployment are unique?

Reply:
Nope, more than one task can be started in the same region/zone. We just need to make sure that no tasks are spilled in regions/zones outside of the ones configured...

Reply:
Ok, I still don't quite get this assertion. Maybe we want to split them up. I see four assertions below. I'm sure these could be their own methods.